src/batList.mli

```diff
@@ -231,6 +231,21 @@ val reduce : ('a -> 'a -> 'a) -> 'a list -> 'a
 
     @raise Invalid_argument on empty list. *)
 
+val fold_left_map : ('a -> 'b -> 'a * 'c) -> 'a -> 'b list -> 'a * 'c list
+(** Combines [fold_left] and [map]. Tail-recursive.
+
+    More precisely :
+
+    [fold_left_map f acc [] = (acc, [])]
+
+    [fold_left_map f acc (x :: xs) =
+    let (acc', y) = f acc x in
+    let (res, ys) = fold_left_map acc' xs in
+    (res, y :: ys)]
```

Comment:
I believe that you should use `{[ ... ]}` for multi-line code in documentation comments.

```
{[
fold_left_map f acc [] = ([], acc)
fold_left_map f acc (x :: xs) =
  let (acc', y) = f acc x in
  let (res, ys) = fold_left_map acc' xs in
  (res, y :: ys)
]}
```

see [batGenlex.mli](https://github.com/ocaml-batteries-team/batteries-included/blob/master/src/batGenlex.mli#L34-L45) for example.